Could not find a declaration file for module 'debug'

## Environment

- wyw-in-js version: 0.5.3
- custom processor: linaria
- Bundler (+ version): TS 5.4.5
- Node.js version: 20.10.0
- OS: Windows 10

## Description

I'm compiling es-modules by Typescript (tsc) in project with React-components which uses linaria:

    typescript@5.4.5
    @linaria/core@6.2.0
    @linaria/react@6.2.1

But I have an error when I try to compile:

    ../../node_modules/@wyw-in-js/shared/types/logger.d.ts:1:26 - error TS7016: Could not find a declaration file for module 'debug'. 'C://myProject/node_modules/debug/src/index.js' implicitly has an 'any' type.
      Try `npm i --save-dev @types/debug` if it exists or add a new declaration (.d.ts) file containing `declare module 'debug';`
    
    1 import genericDebug from 'debug';
                               ~~~~~~~


It seems that @wyw-in-js/shared need to add @types/debug to its dependencies in package.json?

For now I fix the issue by `npm i -D @types/debug`, but I don't think I should have to install the types in my consumer project since I do not directly use package 'debug'.

--- a/.changeset/fix-shared-debug-types.md
+++ b/.changeset/fix-shared-debug-types.md
@@ -1,0 +1,5 @@
+---
+"@wyw-in-js/shared": patch
+---
+
+Fix TypeScript consumer builds by shipping `@types/debug` as a dependency (public `.d.ts` imports `debug`).

--- a/packages/shared/package.json
+++ b/packages/shared/package.json
@@ -2,6 +2,7 @@
   "name": "@wyw-in-js/shared",
   "version": "0.8.1",
   "dependencies": {
+    "@types/debug": "^4.1.9",
     "debug": "^4.3.4",
     "find-up": "^5.0.0",
     "minimatch": "^9.0.3"
@@ -9,7 +10,6 @@
   "devDependencies": {
     "@babel/types": "^7.23.5",
     "@types/babel__core": "^7.20.5",
-    "@types/debug": "^4.1.9",
     "@types/node": "^16.18.55",
     "@wyw-in-js/babel-config": "workspace:*",
     "@wyw-in-js/eslint-config": "workspace:*",

--- a/packages/shared/src/__tests__/dependencies.test.ts
+++ b/packages/shared/src/__tests__/dependencies.test.ts
@@ -1,0 +1,17 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+describe('@wyw-in-js/shared dependencies', () => {
+  it('should ship @types/debug for TypeScript consumers', () => {
+    const dirname = path.dirname(fileURLToPath(import.meta.url));
+    const pkgJsonPath = path.join(dirname, '..', '..', 'package.json');
+    const pkg = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf8')) as {
+      dependencies?: Record<string, string>;
+      devDependencies?: Record<string, string>;
+    };
+
+    expect(pkg.dependencies?.['@types/debug']).toBeTruthy();
+    expect(pkg.devDependencies?.['@types/debug']).toBeUndefined();
+  });
+});

